packages/@react-stately/table/src/Column.ts

```diff
@@ -11,15 +11,16 @@
  */
 
 import {CollectionBuilderContext} from './useTableState';
-import {ColumnProps, TableNode} from '@react-types/table';
+import {ColumnProps} from '@react-types/table';
+import {GridNode} from '@react-types/grid';
 import {PartialNode} from '@react-stately/collections';
 import React, {ReactElement} from 'react';
 
 function Column<T>(props: ColumnProps<T>): ReactElement { // eslint-disable-line @typescript-eslint/no-unused-vars
   return null;
 }
 
-Column.getCollectionNode = function* getCollectionNode<T>(props: ColumnProps<T>, context: CollectionBuilderContext<T>): Generator<PartialNode<T>, void, TableNode<T>[]> {
+Column.getCollectionNode = function* getCollectionNode<T>(props: ColumnProps<T>, context: CollectionBuilderContext<T>): Generator<PartialNode<T>, void, GridNode<T>[]> {
```

Comment:
is there any design doc for this kind of API? I would be very interested in reading through how you have leveraged generators here